src/java.desktop/macosx/classes/sun/lwawt/LWListPeer.java

```diff
@@ -136,16 +136,15 @@ public void removeAll() {
     public void select(final int index) {
         synchronized (getDelegateLock()) {
             getDelegate().setSkipStateChangedEvent(true);
-            getDelegate().getView().setSelectedIndex(index);
+            getDelegate().getView().addSelectionInterval(index, index);
             getDelegate().setSkipStateChangedEvent(false);
         }
     }
 
     @Override
     public void deselect(final int index) {
         synchronized (getDelegateLock()) {
-            getDelegate().getView().getSelectionModel().
-                    removeSelectionInterval(index, index);
+            getDelegate().getView().removeSelectionInterval(index, index);
         }
     }
 
```

Comment:
Note:
`getDelegate().getView().addSelectionInterval(index, index);`
`getDelegate().getView().removeSelectionInterval(index, index);`
just shortcuts for:
`getDelegate().getView().getSelectionModel().addSelectionInterval(index, index);`
`getDelegate().getView().getSelectionModel().removeSelectionInterval(index, index);`